includes/processors/class.llms.processor.course.data.php

```diff
@@ -172,45 +190,109 @@ protected function init() {
 			),
 		);
 
-		// setup throttle vars
-		$this->throttle_max_students = apply_filters( 'llms_data_processor_' . $this->id . '_throttle_count', 2500, $this );
-		$this->throttle_frequency    = apply_filters( 'llms_data_processor_' . $this->id . '_throttle_frequency', HOUR_IN_SECONDS * 4, $this );
+		/**
+		 * Throttles course data processing based on the number of a students in a course.
+		 *
+		 * If the number of students in a course is greater than or equal to this number, the background
+		 * process will be throttled to run only once every N hours where N is equal to the number of hours
+		 * defined by the `llms_data_processor_course_data_throttle_frequency` filter.
+		 *
+		 * @since 3.15.0
+		 * @since [version] Reduced default value of `$number_students` from 2500 to 500.
+		 *
+		 * @see llms_data_processor_course_data_throttle_frequency
+		 *
+		 * @param int                        $number_students The number of students. Default is `500`.
+		 * @param LLMS_Processor_Course_Data $processor       Instance of the data processor class.
+		 */
+		$this->throttle_max_students = apply_filters( 'llms_data_processor_course_data_throttle_count', 500, $this );
+
+		/**
+		 * Throttles course data processing based on the number of a students in a course.
+		 *
+		 * If the number of students in a course is greater than or equal to this number, the background
+		 * process will be throttled to run only once every N hours where N is equal to the number of hours
+		 * defined by the `llms_data_processor_course_data_throttle_frequency` filter.
+		 *
+		 * @since 3.15.0
+		 *
+		 * @see llms_data_processor_course_data_throttle_count
+		 *
+		 * @param int                        $frequency Frequency of the calculation process in seconds. Default `HOUR_IN_SECONDS * 4`.
+		 * @param LLMS_Processor_Course_Data $processor Instance of the data processor class.
+		 */
+		$this->throttle_frequency = apply_filters( 'llms_data_processor_course_data_throttle_frequency', HOUR_IN_SECONDS * 4, $this );
+
+	}
+
+	/**
+	 * Determines if the supplied course is already being processed.
+	 *
+	 * If it's already being processed we'll throttle the processing so we'll wait until the course
+	 * completes it's current data processing and start again later.
+	 *
+	 * @since [version]
+	 *
+	 * @param [type] $course_id [description]
+	 * @return boolean [description]
+	 */
+	protected function is_already_processing_course( $course_id ) {
+
+		if ( ! $this->is_queue_empty() ) {
+			return in_array( $course_id, wp_list_pluck( $this->get_batch()->data, 'post_id' ), true );
```

Comment:
mmm, yea good point, I missed that...

I was thinking about "locking" a course with temporary postmeta data and decided not to because I could pull it this way but I didn't realize it only returned a single batch.

I'll fix it.

Reply:
Most recent push uses course meta data to create (and check) a lock